packages/client/src/components/TopCastDisplay/TopCastDisplay.jsx

```diff
@@ -1,14 +1,39 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { apiURL } from '../../apiURL';
 import {
   Container,
   Typography,
   Avatar,
   Stack,
   Box,
   Button,
+  CircularProgress,
 } from '@mui/material';
 
-const TopCastDisplay = () => {
+const TopCastDisplay = ({ movieID }) => {
+  const [castList, setCastList] = useState([]);
+  const [showFullCast, setShowFullCast] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchCastList = async () => {
```

Comment:
These component were made by different people in our team and then gathered inside this branch to get the impression of how it feels altogether. But now other components are at main already - so probably we will refactor all fetching in a later step (by the way probably it can be a good idea to separate fetching from the components themselves, but its too much for us at the moment)

Reply:
great :) I might keep pointing those out though to stress the need for consistency (and as a reminder haha)